application/src/main/java/net/corda/v5/application/crypto/DigestService.java

```diff
@@ -33,6 +33,22 @@ public interface DigestService {
     @NotNull
     SecureHash hash(@NotNull InputStream inputStream, @NotNull DigestAlgorithmName digestName);
 
+    /**
+     * Parses a secure hash in string form into a {@link SecureHash}.
+     * <p>
+     * A valid secure hash string should be containing the algorithm and hexadecimal representation of the bytes
+     * separated by the colon character (':') ({@link net.corda.v5.crypto.SecureHash.DELIMITER}).
+     * <p>
+     * Please note that this function does not validate the length of the hex string of the digest
```

Comment:
https://github.com/corda/corda-api/blob/release/os/5.0/crypto-extensions/src/main/java/net/corda/v5/crypto/extensions/DigestAlgorithm.java#L59 should ensure that we can carry out that check